Model/TemplateResolver.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Snowdog
+ *
+ * @author      Dawid Czaja <dawid.czaja@snow.dog>.
+ * @category
+ * @package
+ * @copyright   Copyright Snowdog (http://snow.dog)
+ */
+
+namespace Snowdog\Menu\Model;
+
+use Magento\Framework\View\Element\Template;
+
+class TemplateResolver
+{
+    protected $templateMap;
+
+    /**
+     * @param Template $block
+     * @param string $menuId
+     * @param string $oldTemplate
+     * @return string
+     */
+    public function getMenuTemplate($block, $menuId, $oldTemplate)
```

Comment:
I think `$oldTemplate` is not a best choice because it isn't **old** template. Maybe is it better to rename it to `$template`?